tool/checks/driver.dart

```diff
@@ -20,10 +20,10 @@ import 'package:path/path.dart' as path;
 
 import 'rules/visit_registered_nodes.dart';
 
-void main() {
+Future<void> main() async {
```

Comment:
Relaxing `avoid_void_async` to allow this idiom does sort of suggest this Future return type annotation is gratuitous.  Interestingly, this is what the analyzer `ADD_ASYNC` quick-fix produces.  Maybe we should consider updating it to not do this for `main`?

/cc @bwilkerson 

As for a cleanup PR, @albertms10, sure thing.

Thanks for the feedback!

Reply:
I have no idea how common that idiom is, but if it's common enough then it would make sense to support it. Not sure who ought to be looped in to the conversation.